Allow deployment internal mail

ikselven opened the pull request:
Work in progress to implement #1.

What's been done so far?
- mysql backend configuration for postfix is filled in using the MYSQL* environment variables
- SQL queries for mailbox_domains and mailbox_maps against the horde have been put into postfix configuration
- the horde_db container is intialized with a table for the domains
- one domain is inserted (hardcoded though, still needs a variable in .env file)
- entrypoint.sh of postfix image has been extended to expand the configs, see https://github.com/maintaina/containers/commit/9a403b700f89354489ca6b251eb4189e0af68dd7

ikselven committed:
groupware-mail: add mysql backend configuration for postfix

ralflang commented:
Looks promising. As for the user vs mail address problem, multiple solutions are valid:

- use $conf['imap']['maildomain] and fallback to $conf['problems']['maildomain']
- allow any combination of a user from users table and a domain from the virtualdomains table

Later on, mailbox management could be delegated to the vilma app (which is not currently in usable state)

ikselven committed:
groupware-mail: reset mysql postfix configs to postfix defaults

ikselven committed:
groupware-mail: provide template for .env and git ignore .env to prevent accidental leaks

ikselven committed:
groupware-mail: query horde_users in mailbox_maps

- postfix virtual_mailbox_maps query user_uid of horde_users
- Horde users must be named username@domain for this to work

ikselven commented:
Temporary solution for the user vs. mail address problem:
- mysql_virtual_mailbox_maps.cf queries user_uid of table horde_users
- Horde users must be named _username@domain_, e.g. _user1@horde.dev.local_

johanneshuebner committed:
postfix: add domain variable from .env to global

johanneshuebner committed:
add HORDE_DOMAIN to .env.dist

johanneshuebner committed:
rename domaintable.sql file

johanneshuebner committed:
add make script for creating the containers

ikselven committed:
groupware-mail: add setup script configuring submission port and sasl

johanneshuebner committed:
add exit if .env does not exists

johanneshuebner committed:
copy file first to prevent errors

ikselven committed:
groupware-mail: configure smtp for horde

ikselven committed:
groupware-mail: reload postfix at the end, configure dovecot to use only username

ikselven committed:
groupware-mail: comment volumes for extra postfix and dovecot configs

ikselven committed:
Merge branch 'postfix-deployment-internal-mail' into postfix-deployment-internal-mail

ikselven committed:
Merge pull request #3 from johanneshuebner/postfix-deployment-internal-mail

Get mail domain from .env

ralflang commented:
Is this still in do not merge state?

ikselven committed:
groupware-mail: configure container prefix and horde domain from .env values

ikselven committed:
groupware-mail: update README

ikselven committed:
groupware-mail: improve imp config expansion

ikselven committed:
groupware-mail: improve doc of HORDE_DOMAIN and CONTAINER_PREFIX in .env.dist
